std/algorithm/iteration.d

```diff
@@ -876,15 +876,22 @@ template each(alias pred = "a")
                 cast(void)unaryFun!pred(a);
         }));
 
+    enum isForeachUnaryWithIndexIterable(R) =
+        is(typeof((R r) {
+            foreach (i, ref a; r)
+                cast(void)unaryFun!pred(i, a);
+        }));
+
     enum isForeachBinaryIterable(R) =
         is(typeof((R r) {
-            foreach (ref i, ref a; r)
-                cast(void)binaryFun!BinaryArgs(i, a);
+            foreach (ref a, ref b; r)
+                cast(void)binaryFun!BinaryArgs(a, b);
```

Comment:
To symbolize that it's a binary iteration with two elements - `i` is usually used as designated variable for the loop index.
The idea was to make it similar to `std.functional.binaryFun`.